### ISSUE TITLE
[4.0] [bug] Joomla does not destroy the session when logging out with any active cache option

### ISSUE THREAD

**Issue by Stuartemk** [joomla/joomla-cms#35971]
### Steps to reproduce the issue
I have my site joomla 4.0.4 through Cloudflare, and I realized that when logging in, logging out and reopening it sent invalid token error (white screen of death) which generated many problems, however I changed the session management before the database to REDIS, and the problem continued, so obviously leaving the session out the problem was not with the Cloudflare configuration, so doing a lot of testing, I could find that the error originates:
1.- When the cache plugin is enabled and the main configuration without cache enabled.
2.- When the main cache configuration is enabled without the cache plugin enabled.
3.- When the main configuration cache is enabled and the cache plugin enabled.

In conclusion, any enabled cache option causes the invalid token error (even managing sessions with REDIS)
I have enabled the back of the balancer and I even checked and version 4.0.4 already has the lines removed https://github.com/joomla/joomla-cms/pull/35509#issuecomment-953512918

### Expected result

When closing the session, the token must be destroyed with any enabled chaché configuration enabled,

### Actual result

4.0 Joomla does not destroy the session when logging out with any active cache option

### Additional comments
Possibly this should be marked as a release blocker
